contracts/settling_game/modules/crypts/CryptRun.cairo

```diff
@@ -0,0 +1,277 @@
+
+    if (is_seed_active == FALSE) {
+        // get random
+        tempvar randomSeed = 123456842657854254785;
+
+        // write crypt set ts 1 day from now
+        crypt.write(crypt_id, Crypt(now + DAY, randomSeed, NUMBER_SUCCESSFUL_RUNS));
+
+        // write adventurer
+        adventurer.write(
+            adventurer_id, AdventurerLocation(crypt_id.low, NODE_START_INDEX, randomSeed)
+        );
+
+        return (seed=randomSeed);
+    }
+
+    // we save the seed within the adventurer struct, this is incase the seed changes midrun the player can still complete
+    adventurer.write(
+        adventurer_id, AdventurerLocation(crypt_id.low, NODE_START_INDEX, crypt_status.seed)
+    );
+
+    return (seed=crypt_status.seed);
+}
+
+// @notice Move Adventurer within the Crypt.
+// @param crypt_id: Crypts erc721 ID
+// @param adventurer_id: Adventurer erc721 ID
+// @param start_location: Index of the current location
+// @param end_location: Index of the location wishing to travel to
+@external
+func move_to_location_in_crypt{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    crypt_id: Uint256, adventurer_id: Uint256, start_location: felt, end_location: felt
+) -> () {
+    alloc_locals;
+    // check ownership
+    assert_adventurer(adventurer_id);
+
+    // assert at location requested
+    let (adventurer_location) = assert_and_return_location(crypt_id, adventurer_id, start_location);
+
+    let (dungeon) = build_dungeon(crypt_id, adventurer_location.seed);
+
+    // check path exists
+    let (shortest_path_len, identifiers, total_distance) = Crypts.check_path_exists(
+        dungeon, start_location, end_location
+    );
+
+    // check path exists
+    with_attr error_message("Crypts: unmoveable location from your location") {
+        assert_not_zero(shortest_path_len);
+    }
+
+    // move player and store state
+    adventurer.write(
+        adventurer_id, AdventurerLocation(crypt_id.low, end_location, adventurer_location.seed)
+    );
+
+    return ();
+}
+
+// @notice Can be called when the Adventurer exists on a Vertex
+// @param crypt_id: Crypts erc721 ID
+// @param adventurer_id: Adventurer erc721 ID
+// @param location: Index of the vertex
+@external
+func interact_with_vertex{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    crypt_id: Uint256, adventurer_id: Uint256, location: felt
+) -> () {
+    alloc_locals;
+    // check ownership
+    assert_adventurer(adventurer_id);
+
+    // assert at location
+    let (adventurer_location) = assert_and_return_location(crypt_id, adventurer_id, location);
+
+    // build dungeon
+    let (dungeon) = build_dungeon(crypt_id, adventurer_location.seed);
+
+    // get entites
+    let (entity_list_len, entity_list) = Crypts.get_entity_list(dungeon, adventurer_location.seed);
+
+    // check entity exists
+    let (entity_exists) = Crypts.check_entity_at_index(location, entity_list_len, entity_list);
```

Comment:
This function could actually do the `assert` internally, that's how it's used anyway and its comment also mentions asserting :)